libs/librepcb/projecteditor/boardeditor/boardeditor.cpp

```diff
@@ -533,6 +535,53 @@ void BoardEditor::on_actionExportAsPdf_triggered() {
   }
 }
 
+void BoardEditor::on_actionExportAsSvg_triggered() {
+  try {
+    Board* board = getActiveBoard();
+    if (!board) {
+      throw Exception(__FILE__, __LINE__, tr("No board selected."));
+    }
+    QString projectName =
+        FilePath::cleanFileName(*mProject.getMetadata().getName(),
+                                FilePath::ReplaceSpaces | FilePath::KeepCase);
+    QString projectVersion =
+        FilePath::cleanFileName(mProject.getMetadata().getVersion(),
+                                FilePath::ReplaceSpaces | FilePath::KeepCase);
+    QString relativePath =
+        QString("output/%1/%2_Board.svg").arg(projectVersion, projectName);
+    FilePath defaultFilePath = mProject.getPath().getPathTo(relativePath);
+    QDir().mkpath(defaultFilePath.getParentDir().toStr());
+    QString filename = FileDialog::getSaveFileName(
+        this, tr("SVG Export"), defaultFilePath.toNative(), "*.svg");
```

Comment:
I always liked the UX when exporting images in GIMP (`File » Export As...`), where you can change the file name to have a different extension (or choose a file type from the list) and then, depending on the selected file type, you where presented with a 2nd dialog with configuration for your format (for changing the DPI and what not), if the format supported any configuration.

P.S. Thanks for all the great work on this.

Reply:
Yeah I agree regarding the "Export As...", actually I considered to do it that way too. But since SVG is currently the only option, I decided to create an explicit menu item instead, but we can change it to a more generic "Export As..." menu item once we support more export formats.

If there is a strong opinion, I could change it now - otherwise I would change it when it's needed :)